recaf-launcher/src/main/java/dev/xdark/recaf/Launcher.java

```diff
@@ -0,0 +1,251 @@
+    parser.allowsUnrecognizedOptions();
+    OptionSet options = parser.parse(args);
+    Path jarPath = options.valueOf(jarOption);
+    boolean exists = Files.exists(jarPath), valid = exists;
+    log("Target jar exists: %s", exists);
+    if (info == null) {
+      // We were unable to fetch latest release info.
+      // Attempt to continue.
+      if (valid) {
+        attemptLaunch(jarPath, UPDATE_FAIL);
+      } else {
+        log("Launcher was unable to fetch release info, cannot continue.");
+        log("If you believe that it is a bug, please open an issue at: ");
+        log(ISSUES_URL);
+        System.exit(1);
+      }
+    }
+    boolean autoUpdate = options.valueOf(autoUpdateOption);
+    log("Automatic updates: %s", autoUpdate);
+    if (valid) {
+      String version = null;
+      try (JarFile jar = new JarFile(jarPath.toFile())) {
+        Manifest manifest = jar.getManifest();
+        version = manifest.getMainAttributes().getValue("Specification-Version");
+        log("Detected current version: %s", version);
+      } catch (IOException ex) {
+        log("Unable to get Recaf version: ", ex);
+      }
+      if (version == null) {
+        log("Version detected has failed");
+        log("Ensure that Recaf's jar file is not damaged, or");
+        log("Open an issue if you think that it's an error: ");
+        log(ISSUES_URL);
+        valid = false;
+      } else {
+        try {
+          valid = !isOutdated(version, info.getName());
+        } catch (Exception ex) {
+          log("Could not compare versions: ", ex);
+          valid = false;
+        }
+      }
+    }
+    Asset asset = findReleaseAsset(info.getAssets());
+    if (asset == null) {
+      log("Unable to detect release asset");
+    }
+    if (valid) {
+      if (asset == null) {
+        log("Launcher was unable to detect release asset from GitHub releases");
+        log("Please open an issue at: ");
+        log(ISSUES_URL);
+        attemptLaunch(jarPath, UPDATE_FAIL, 1);
+      }
+      try {
+        valid = asset.getSize() == Files.size(jarPath);
+        log("Jar size match: %s", valid);
+      } catch (IOException ex) {
+        log("Could not verify jar size, will assume that jar is valid: ", ex);
+      }
+    }
+    if (!valid) {
+      if (exists) {
+        boolean writeable = Files.isWritable(jarPath);
+        if (!writeable) {
+          log("Jar is not writeable, check your file system permissions");
+          attemptLaunch(jarPath, UPDATE_FAIL, 1);
+        }
+      }
+      // Download new jar.
+      String $url = asset.getUrl();
+      log("Downloading update from: %s", $url);
+      byte[] bytes;
+      try {
+        URL url = new URL($url);
+        try (InputStream in = url.openStream()) {
+          bytes = IOUtils.toByteArray(in);
+        }
+        log("Downloaded updated jar, starting");
+        Files
+            .write(jarPath, bytes, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
```

Comment:
Is this supposed to update the contents of the jar? I believe this is a raw bytes truncate instead, and it will just corrupt it. Would have to test to verify.

Reply:
`TRUNCATE_EXISTING` will truncate existing file, so bytes do not append.